### ISSUE TITLE
Compatibility issues with Majrusz's Progressive Difficulty

### ISSUE THREAD

**Issue by Blacksky649** [someaddons/performant_issues#147]
Skeleton AI during Undead Army raid's seems to only focus on reaching its moving goal and do not attack the player while using performant. This does not apply to zombies in the raid who operate normally (moving goal and attack players). I tried turning all performant features off in the config and no change in the behaviour.

A quick observation I made is that if I leave the map and load back in that the skeletons start attacking me again and have normal behaviour.

version: 1.16.5

**Comment by Majrusz** [comment]
Hi, Progressive Difficulty creator here. I managed to fix this compatibility issue with my mod, but the problem still exists. What I mean is that there is a problem with spawning any skeleton-like enemy and then giving them any piece of armor. This code will break their AI:
```
SkeletonEntity skeleton = EntityType.SKELETON.spawn( world, null, null, null, position, SpawnReason.EVENT, true, true );
if( skeleton != null )
    skeleton.setItemStackToSlot( EquipmentSlotType.HEAD, new ItemStack( Items.LEATHER_HELMET, 1 ) );
```

**Comment by rtxyd** [comment]
Same problem when I load with "improvedmobs". It makes skeletons not attack me.

**Comment by someaddons** [comment]
fixed in 3.57, thanks for the reproduce!
